### PR TITLE
impl. e2e test for system transactions

### PR DESCRIPTION
**Pull request by eyusufatik**

### CODE REVIEW

File: bin/citrea/tests/e2e/mod.rs [hunk 1]
```diff
@@ -1316,3 +1318,146 @@ async fn test_reopen_prover() -> Result<(), anyhow::Error> {
...
+                *hex::decode(
+                    "1f5783330000000000000000000000000000000000000000000000000000000000000003"
+                )
+                .unwrap()
+                .as_slice()
+            );
+
+            assert_eq!(
+                set_tx.from,
+                Address::from_str("0xdeaddeaddeaddeaddeaddeaddeaddeaddeaddead").unwrap()
```
Comment (kpp) [inline bin/citrea/tests/e2e/mod.rs:1382]: address!("deaddeaddeaddeaddeaddeaddeaddeaddeaddead")

**Review by kpp — DISMISSED** [review]
Minor change requested, but LGTM

File: bin/citrea/tests/e2e/mod.rs (hunk 1 above)
Comment (eyusufatik) [reply bin/citrea/tests/e2e/mod.rs:1382]: those are different Address's from different crates unfortunately 😓 

but I'll push a change with less code dupe.

File: bin/citrea/tests/e2e/mod.rs [hunk 2]
```diff
@@ -1316,3 +1318,134 @@ async fn test_reopen_prover() -> Result<(), anyhow::Error> {
...
+            assert_eq!(set_tx.to.unwrap(), system_contract_address);
+            assert_eq!(
+                set_tx.input[0..4],
+                *hex::decode("0e27bc11").unwrap().as_slice()
+            );
+        } else {
+            assert_eq!(block.transactions.len(), 1);
+
+            let tx = &block.transactions[0];
+            // maybe use SYSTEM_SIGNER and L1BlockHashList::address() here?
```
Comment (kpp) [inline bin/citrea/tests/e2e/mod.rs:1390]: I think no need

**Review by kpp — APPROVED** [review]

**Review by kpp — APPROVED** [review]

**Pull request merged**